wavelength-plugins/lottery.js

```diff
@@ -0,0 +1,121 @@
+/****************************************
+ * Lottery Plug-in for Pokémon Showdown
+ *            Created by:
+ *         HoeenHero and Insist
+ ****************************************/
+
+"use strict";
+
+class Lottery {
+	constructor(room, user) {
+		this.players = Object.create(null);
+		this.room = room;
+		room.lottoNumber = room.lottoNumber ? room.lottoNumber++ : 1;
+		this.lottoNumber = room.lottoNumber;
+		this.costToJoin = 3;
+		this.state = "signups";
+		this.playerCount = 0;
+		this.room.add(`|uhtml|lottery-${this.lottoNumber}|<div class="broadcast-blue"><p style="font-size: 14pt; text-align: center">A new <strong>Lottery drawing</strong> is starting!</p><p style="font-size: 9pt; text-align: center"><button name="send" value="/lotto join">Join</button><br /><strong>DISCLAIMER: Joining costs ${this.costToJoin} ${currencyPlural}!!!!</strong></p></div>`, true);
+		this.timer = setTimeout(() => {
+			if (this.playerCount < 2) {
+				this.room.add('|uhtmlchange|lottery-' + this.lottoNumber + '|<div class="broadcast-red"><p style="text-align: center; font-size: 14pt>This Lottery drawing has ended due to lack of users.</p></div>');
+				return this.end();
+			}
+			this.drawWinner();
+		}, 1000 * 60 * 60 * 24);
+	}
+
+	onConnect(user, connection, room) {
+		if (this.state === "signups") {
+			user.sendTo(this.room, '|uhtml|lottery-' + this.lottoNumber + '|<div class="broadcast-blue"><p style="text-align: center; font-size: 14pt>A Lottery Drawing has started looking for players!<hr /><br />For the price of 3 ' + currencyPlural + ', you can earn 5 ' + currencyPlural + ' plus one ' + currencyName + ' per user who joins.</p><br /><button name="send" value="/lottery join">Click here to join the Lottery</button></div>');
+		} else {
+			user.sendTo(this.room, '|uhtmlchange|lottery-' + this.lottoNumber + '|<div class="broadcast-green"><p style="text-align: center; font-size: 14pt>This Lottery drawing is about to announce a winner!!!');
+		}
+	}
+
+	drawWinner(user) {
+		let basePrize = 5;
+		let lottoPrize = basePrize + this.playerCount + this.costToJoin;
+		let winner = toId(user.name);
```

Comment:
This is completely wrong. You need to pick a winner out of the participants not win whoever was passed to this function. (which in its current state will either crash the server, or win whoever used the command to end the lottery early.)